__tests__/Guides.unit.test.ts

```diff
@@ -1,10 +1,560 @@
+      (guidesInstance as any).selectGuide(selectedGuide, eventMouse as any);
+      expect(guidesInstance.state.selectedGuides).toEqual([]);
+    });
+
+    it('should reset a selected guide', () => {
+      (guidesInstance as any).selectGuide(selectedGuide, eventMouse as any);
+      guidesInstance.resetSelected();
+      expect(guidesInstance.state.selectedGuides).toEqual([]);
+    });
+
+    it('should return an index of selected guide', () => {
+      (guidesInstance as any).selectGuide(selectedGuide, eventMouse as any);
+      expect((guidesInstance as any).getSelectedGuideIndex(guidesTest)).toEqual(1);
+    });
+
+    it('should delete a selected guide', () => {
+      guidesInstance.loadGuides(guidesTest);
+      const newGuides = guidesInstance.getGuides().slice();
+      newGuides.splice(indexOfDeletedGuide, 1);
+      const keyboardEvent = new KeyboardEvent('keydown', {
+        code: 'Backspace',
+      });
+      (guidesInstance as any).selectGuide(deletedGuide, eventMouse as any);
+      guidesInstance.deleteSelectedGuide(keyboardEvent);
+      expect(guidesInstance.state.selectedGuides).toEqual([]);
+      expect(guidesInstance.getGuides()).toEqual(newGuides);
+    });
+
+    it("shouldn't delete a selected guide because of incorrect button's click", () => {
+      guidesInstance.loadGuides(guidesTest);
+      const keyboardEvent = new KeyboardEvent('keydown', {
+        code: 'Enter',
+      });
+      (guidesInstance as any).selectGuide(deletedGuide, eventMouse as any);
+      guidesInstance.deleteSelectedGuide(keyboardEvent);
+      expect(guidesInstance.state.selectedGuides).toEqual([deletedGuide]);
+      expect(guidesInstance.getGuides()).toEqual(guidesTest);
+    });
+
+    it("shouldn't delete a selected guide because any guide isn't selected", () => {
+      guidesInstance.loadGuides(guidesTest);
+      const keyboardEvent = new KeyboardEvent('keydown', {
+        code: 'Backspace',
+      });
+      guidesInstance.deleteSelectedGuide(keyboardEvent);
+
+      expect(guidesInstance.state.selectedGuides).toEqual([]);
+      expect(guidesInstance.getGuides()).toEqual(guidesTest);
+    });
+  });
+
+  it("should return a name of class, given that there's not a selected guide", () => {
+    expect((guidesInstance as any).currentGuideClassName(selectedGuide)).toContain(defaultProps.type);
+    expect((guidesInstance as any).currentGuideClassName(selectedGuide)).not.toContain('selected');
+  });
+
+  it("should return a name of class, given that there's a selected guide", () => {
+    (guidesInstance as any).selectGuide(selectedGuide, eventMouse as any);
+    expect((guidesInstance as any).currentGuideClassName(selectedGuide)).toContain(defaultProps.type);
+    expect((guidesInstance as any).currentGuideClassName(selectedGuide)).toContain('selected');
+  });
+
+  it('should calculate the transform property of style for a horizontal guide', () => {
+    expect((guidesInstance as any).calcHorizontalTransform(4)).toMatch(/-21/);
```

Comment:
add moreexpects here with different parameters to make sure it is works for 100%